core/dive.c

```diff
@@ -246,9 +246,11 @@ void add_extra_data(struct divecomputer *dc, const char *key, const char *value)
 enum divemode_t get_current_divemode(const struct divecomputer *dc, int time, const struct event **evp, enum divemode_t *divemode)
 {
 	const struct event *ev = *evp;
-	if (*divemode == UNDEF_COMP_TYPE && dc) {
-		*divemode = dc->divemode;
-		ev = get_next_event(dc->events, "modechange");
+	if (dc) {
+		if (*divemode == UNDEF_COMP_TYPE) {
+			*divemode = dc->divemode;
+			ev = get_next_event(dc->events, "modechange");
+		}
```

Comment:
so with this new code in case `dc != NULL && *divemode != UNDEF_COMP_TYPE` we don't assign `ev` at all. Is that what you want?

Reply:
Yes, that is exactly the case that needed fixing. It reuses the ev from the previous invocation. The first invocation should be with *divemode set to UNDEF_COMP_TYPE.